src/duco/ducobox.py

```diff
@@ -719,15 +718,15 @@ def execute_command(self, command):
         logging.debug('Serial command:\n{command}'.format(command=command))
         reply = ''
         if self._serial:
-            self._serial.write('\r')
+            self._serial.write('\r'.encode())
```

Comment:
this should be the fix, the rest of this MR is some fixes for static checker and python versions support

Reply:
Confirmed, this prevents the error from occurring.